lib/string_segmenter.rb

```diff
@@ -1,2 +1,43 @@
+require 'dictionary'
+
 def segment_string(str)
-end
\ No newline at end of file
+#1. create an array to store words
+	arr = []
+#2. loop over the first character and check if it's a word according to dictionary.rb
+	str.each do |n|
+	#check if character 1 is word
+	if valid_word?(n)
+		#if it is, store it in the last place of the array
+			arr[-1] << n
+	#if it's not a word, add the next character in the string and check again
+	elsif !valid_word?(n)
+		n=n+n(1)
```

Comment:
`n(1)` looks like a method. Usually when we see a "keyword" (which is basically any non-quoted word) it means it's either a variable or method. And a keyword immediately followed by `(` means it's a method as opposed to a variable.

So what is your _goal_ her? Is it to set `n` as `n` plus `1`? Or something else?